dex/order/order_test.go

```diff
@@ -671,3 +699,40 @@ func TestUnixTimeMilli(t *testing.T) {
 		})
 	}
 }
+
+func TestPreimage_Commit(t *testing.T) {
+	tests := []struct {
+		name string
+		pi   *Preimage
+		want Commitment
+	}{
+		{
+			"ok zeros",
+			&Preimage{},
+			Commitment{
+				0x05, 0xf6, 0xac, 0x47, 0xac, 0xcd, 0x33, 0x8d, 0x32, 0x9c, 0xc1,
+				0x6f, 0x6d, 0x59, 0xf3, 0x40, 0x9c, 0xc8, 0xbf, 0xe7, 0x6a, 0x27,
+				0x2e, 0x1e, 0xec, 0x61, 0x2e, 0x49, 0xc1, 0x15, 0x14, 0x5d},
+		},
+		{
+			"ok one bit",
+			&Preimage{0x01},
+			Commitment{
+				0x8e, 0xa4, 0x09, 0x18, 0xf0, 0x47, 0x2d, 0xdd, 0xdd, 0x8e, 0xe0,
+				0x6f, 0xab, 0xfe, 0xbc, 0xde, 0xca, 0x0c, 0xad, 0x2f, 0xe4, 0xa0,
+				0x69, 0xc7, 0xe4, 0x17, 0x2b, 0x81, 0x9c, 0x2e, 0xe5, 0x07},
```

Comment:
We'll likely want to prohibit use of stupid or zero preimages (definitely commitments since those would be indicate the user doesn't actually know the preimage), if not reuse of preimages.